.github/workflows/doc-tests.yaml

```
@@ -2,20 +2,33 @@ name: Lint (Docs)
 run-name: Lint (Docs)
 on:
   pull_request:
-    paths:
-      - 'CHANGELOG.md'
-      - 'docs/**'
-      - 'examples/**'
+
   merge_group:
-    paths:
-      - 'docs/**'
-      - 'examples/**'
-      - 'CHANGELOG.md'
+
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      changed: ${{ steps.changes.outputs.changed }}
+    steps:
+      - uses: dorny/paths-filter@4067d885736b84de7c414f582ac45897079b0a78
+        id: changes
+        with:
+          base: ${{ github.event.pull_request.base.ref || github.event.merge_group.base_ref }}
+          ref: ${{ github.event.pull_request.head.ref || github.event.merge_group.head_ref }}
+          filters: |
+            changed:
+              - 'CHANGELOG.md'
+              - 'docs/**'
+              - 'examples/**'
```

```suggestion
            changed:
              - .github/workflows/doc-tests.yaml
              - 'CHANGELOG.md'
              - 'docs/**'
              - 'examples/**'
```

Currently this check is skipped when this file changes:

https://github.com/gravitational/teleport/actions/runs/6202035532/job/16839948523?pr=31953#step:2:26

It doesn't matter for this case, since we test the important part for *this* pr.  But we'll want to test other changes that occur later in the workflow.